### PR TITLE
Lazy sanitization of SQL, and a couple of performance improvements

### PR DESCRIPTION
**Pull request by itsderek23**
Previously, we were sanitizing SQL queries as layers were added. Now, we sanitize only where it is needed: `LayerSlowTransactionConverter`. This results in large performance gains:

* overhead benchmarks drop from 8.1% to 3.2%
* stackprof samples for `ScoutApm::Utils::SqlSanitizer#to_s_postgres` drop from 1.2% to 0%

This also addresses a couple other higher-sample % numbers from stackprof output:

* Memo-izing fetching the database driver
* Using strip! vs. gsub!

This needs to run on production environments first before merging.

### CODE REVIEW

File: lib/scout_apm/utils/sql_sanitizer.rb [hunk 1]
```diff
@@ -57,7 +60,8 @@ def to_s_sqlite
         sql.gsub!(SQLITE_REMOVE_STRINGS, '?')
         sql.gsub!(SQLITE_REMOVE_INTEGERS, '?')
```
Comment (cschneid) [inline lib/scout_apm/utils/sql_sanitizer.rb]: If you want to get fancy, you can do this in a single regex pass by doing:

```
sql.gsub!(SQLITE_REMOVE_STRINGS.union(SQL_REMOVE_INTEGERS), '?')
```

No clue if its faster, but it'd be one less call to `gsub!`

**Comment by cschneid** [comment]
:+1:

**Comment by itsderek23** [comment]
This has been running on prod for a while and looking good. Bumped version to 1.2.12 and updated changelog.

**Comment by itsderek23** [comment]
> Bumped version to 1.2.12 and updated changelog.

Actually 1.2.13. 1.2.12 was already released.

**Pull request merged**